### PR TITLE
III-6117 BUGFIX put saved search gave error when saving twice

### PR DESCRIPTION
**Pull request by grubolsch**
### Fixed
Fixed a bug with the PUT of saved searches.
If you saved a saved search without modifying any value, it would throw an `ApiProblem::savedSearchNotFound()`.

This is because I listened to the "affected rows", which will return 0 if the update does not have to change anything in the db.

I now use a proper select query.

I provided an extra feature test to prove the bug is fixed.

---
Ticket: https://jira.uitdatabank.be/browse/III-6117

### CODE REVIEW

**Review by JonasVHG — APPROVED** [review]

File: src/SavedSearches/WriteModel/SavedSearchRepositoryInterface.php [hunk 1]
```diff
@@ -20,10 +21,12 @@ public function update(
...
         QueryString $queryString
-    ): int;
+    ): void;
 
     public function delete(
         string $userId,
         string $searchId
     ): void;
+
+    public function findSavedSearchOwnedByCurrentUser(string $id): ?SavedSearch;
```
Comment (LucWollants) [inline src/SavedSearches/WriteModel/SavedSearchRepositoryInterface.php:31]: This interface is confusing me, all the other interfaces pass in an `id` and a `userId`. I think it's best to stick with that approach and change the method to `findByIdAndUserId`.

I would also remove the `userId` property from the constructor of `UDB3SavedSearchRepository`

File: features/saved_searches/update.feature [hunk 2]
```diff
@@ -33,8 +33,33 @@ Feature: Test the UDB3 saved searches API
...
+#    And the response body should be valid JSON
     And the JSON response should include:
       """
       {"name":"%{name}","query":"Lessen in de avond","id":"%{id}"}
-      """
\ No newline at end of file
+      """
+
+  #Bugfix https://jira.publiq.be/browse/III-6019
+  Scenario: I update a saved search twice
```
Comment (LucWollants) [inline features/saved_searches/update.feature:43]: This makes me wonder if we need to make a test where one user would be updating the saved search of another user?

File: src/SavedSearches/WriteModel/SavedSearchRepositoryInterface.php (hunk 1 above)
Comment (LucWollants) [reply src/SavedSearches/WriteModel/SavedSearchRepositoryInterface.php:31]: I now see that this new method is inside the wrong interface, there is a read and a write interface. This method should be part of `CultuurNet\UDB3\SavedSearches\ReadModel\SavedSearchRepositoryInterface`

**Review by LucWollants — CHANGES_REQUESTED** [review]
The new method is not inside the correct interface, it should be added to `CultuurNet\UDB3\SavedSearches\ReadModel\SavedSearchRepositoryInterface`

File: src/SavedSearches/WriteModel/SavedSearchRepositoryInterface.php (hunk 1 above)
Comment (grubolsch) [reply src/SavedSearches/WriteModel/SavedSearchRepositoryInterface.php]: Moved it to a separate SavedSearchReadRepository class.

File: features/saved_searches/update.feature (hunk 2 above)
Comment (grubolsch) [reply features/saved_searches/update.feature:43]: Added a test for this, and also for when the saved search does not exist

**Comment by grubolsch** [comment]
I processed all feedback, and added some new feature / unit tests.
Also rerun all features test related to saved searches.

**Review by LucWollants — APPROVED** [review]
Nice work!

**Pull request merged**